operator/v1/types_metal3provisioning.go

```diff
@@ -0,0 +1,109 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Metal3Provisioning contains configuration used by the Provisioning
+// service (Ironic) to provision baremetal hosts.
+//
+// Metal3Provisioning is created by the Openshift installer using admin
+// or user provided information about the provisioning network and the NIC
+// on the server that can be used to PXE boot it.
+//
+// This CR is a singleton, created by the installer and currently only
+// consumed by the machine-api-operator to bring up and update containers
+// in a metal3 cluster.
+//
+type Metal3Provisioning struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the
+	// Metal3Provisioning.
+	Spec Metal3ProvisioningSpec `json:"spec,omitempty"`
+
+	// status is the most recently observed status of the
+	// Metal3Provisioning.
+	Status Metal3ProvisioningStatus `json:"status,omitempty"`
+}
+
+// ProvisioningDHCP represents just the configuration required to fully
+// identify the way DHCP would be handled during baremetal host bringup.
+//
+// DHCP services could be provided external to the metal3 cluster, in
+// which case, IP address assignment for the baremetal servers should
+// happen via this external DHCP server and not via a DHCP server started
+// within the metal3 cluster.
+// If IP address assignment needs to happen via the DHCP server within the
+// metal3 cluster, then the CR needs to contain the DHCP address range that
+// this internal DHCP server needs to use.
+//
+type ProvisioningDHCP struct {
```

Comment:
I really don't understand the doc on this field.  Is this controlling what will be (desired state) or reflecting what is (actual state).

I'm going to try to find @sadasu for a bluejeans chat.

Reply:
I replied above already, this describes the desired state of the DHCP services run on the cluster.

This is configuration data which we use to configure a DHCP server that's hosted on the cluster, for PXE boot provisioning of baremetal servers.

Reply:
In order to configure a DHCP server, you have to specify a pool or IP range for the leases, that's what this provides - @deads2k how can we adjust the comment to explain this more clearly (to me it already seems clear FWIW), any suggestions?